### PR TITLE
[docs] Deploying ray on a cluster

### PR DESCRIPTION
**Pull request by wuisawesome**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/commands.rst [hunk 1]
```diff
@@ -136,7 +136,7 @@ To download or upload files to the cluster head node, use ``ray rsync_down`` or
 
 .. _monitor-cluster:
 
-Monitoring cluster status (``ray dashboard/monitor``)
+Monitoring cluster status (``ray dashboard/status``)
 -----------------------------------------------------
 
 The ray also comes with an online dashboard. The dashboard is accessible via HTTP on the head node (by default it listens on ``localhost:8265``). You can also use the built-in ``ray dashboard`` to do this automatically.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/commands.rst:142]: Unrelated to this PR, but could you fix "The ray" to "Ray"

File: doc/source/cluster/commands.rst [hunk 2]
```diff
@@ -145,12 +145,17 @@ The ray also comes with an online dashboard. The dashboard is accessible via HTT
 
     $ ray dashboard cluster.yaml
 
-You can monitor cluster usage and auto-scaling status by tailing the autoscaling
-logs in ``/tmp/ray/session_*/logs/monitor*``.
+You can monitor cluster usage and auto-scaling status by running
```
Comment (DmitriGekhtman) [inline doc/source/cluster/commands.rst:148]: Could you indicate here that `ray status` has to be run on a Ray node?

File: doc/source/cluster/guide.rst [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+Ray is designed to support highly elastic workloads which are most efficient on
+an autoscaling cluster. At a high level, the autoscaler attempts to
+launch/terminate nodes in order to ensure that workloads have sufficient
+resources to run, while minimizing the idle resources.
+
+It does this by taking into consideration:
+
+* User specified hard limits (min/max workers).
+* User specified node types (nodes in a ray cluster do _not_ have to be
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:60]: nit: capitalize Ray?

File: doc/source/cluster/guide.rst [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+address of the ray client server.
+
+:ref:`Learn more about setting up the ray client server here <ray-client>`.
+
+.. note::
+
+  When deploying an application, the job will be killed if the driver
+  disconnects.
+
+  A detached actor can be used to avoid having a long running driver.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:115]: Links to docs on detached actors / drivers could help if these docs exist.

File: doc/source/cluster/guide.rst [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+``http://localhost:8265``.
+
+With the kubernetes operator, you will need to expose port 8265 on the head
+node, or use `kubectl to portforward
+<https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/>`__.
+
+Observing the autoscaler
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Recall that the autoscaler makes decisions by considering the user defined
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:147]: nit: Could remove "recall that", too academic.

side: I've seen mathematicians say "Recall if you don't already know"

File: doc/source/cluster/guide.rst [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+configuration, scheduling information, and programatic information from the
+cluster. The autoscaler publishes its inputs from the cluster, the state of
+nodes being launched, and any errors generated, which can be accessed via the
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:150]: Could split the sentence up a bit

"; all of this information can be accessed with..."

File: doc/source/cluster/guide.rst [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+defined as :ref:`a helm chart <ray-helm>` and will:
+
+* Create a controller (the operator).
+* Provision a new pod (head node).
+* execute shell commands to set up Ray with the provided options.
+* (optionally) run any custom, user defined setup commands.
+* Initialize the ray cluster.
+
+The operator will then serve as the autoscaler.
+
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:48]: This description is mostly correct but the details are subject to change in the near future.

Mind if I open a PR against your branch?

**Review by DmitriGekhtman — COMMENTED** [review]
This is awesome, some comments.

File: doc/source/cluster/guide.rst (hunk 7 above)
Comment (wuisawesome) [reply doc/source/cluster/guide.rst:48]: go for it (you can also directly edit this branch)

**Comment by richardliaw** [comment]
@wuisawesome your target branch is not master; can you close this PR and make a new one?

**Comment by wuisawesome** [comment]
Yup, #16538

**Pull request closed without merging**